File: go-client-sdk/pkg/models/operations/patchcustomerscustomeridusage.go
package operations

import (
	"time"
)

type PatchCustomersCustomerIDUsagePathParams struct {
	CustomerID string `pathParam:"style=simple,explode=false,name=customer_id"`
}

type PatchCustomersCustomerIDUsageQueryParams struct {
	TimeframeEnd   time.Time `queryParam:"style=form,explode=true,name=timeframe_end"`
	TimeframeStart time.Time `queryParam:"style=form,explode=true,name=timeframe_start"`
}

type PatchCustomersCustomerIDUsageRequestBodyEvents struct {
	EventName  string                 `json:"event_name"`
	Properties map[string]interface{} `json:"properties"`
	Timestamp  string                 `json:"timestamp"`
}

type PatchCustomersCustomerIDUsageRequestBody struct {
	Events []PatchCustomersCustomerIDUsageRequestBodyEvents `json:"events"`
}

type PatchCustomersCustomerIDUsage200ApplicationJSON struct {
	Duplicate []map[string]interface{} `json:"duplicate,omitempty"`
	Ingested  []string                 `json:"ingested,omitempty"`
}

type PatchCustomersCustomerIDUsage400ApplicationJSONValidationErrors struct {
	IdempotencyKey   *string  `json:"idempotency_key,omitempty"`
	ValidationErrors []string `json:"validation_errors,omitempty"`
}

type PatchCustomersCustomerIDUsage400ApplicationJSON struct {
	Status           int64                                                             `json:"status"`
	Title            string                                                            `json:"title"`
	Type             string                                                            `json:"type"`
	ValidationErrors []PatchCustomersCustomerIDUsage400ApplicationJSONValidationErrors `json:"validation_errors"`
}

type PatchCustomersCustomerIDUsageRequest struct {
	PathParams  PatchCustomersCustomerIDUsagePathParams
	QueryParams PatchCustomersCustomerIDUsageQueryParams
	Request     *PatchCustomersCustomerIDUsageRequestBody `request:"mediaType=application/json"`
}

type PatchCustomersCustomerIDUsageResponse struct {
	ContentType                                           string
	StatusCode                                            int64
	PatchCustomersCustomerIDUsage200ApplicationJSONObject *PatchCustomersCustomerIDUsage200ApplicationJSON
	PatchCustomersCustomerIDUsage400ApplicationJSONObject *PatchCustomersCustomerIDUsage400ApplicationJSON
}

File: typescript-client-sdk/src/sdk/models/operations/patchcustomerscustomeridusage.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform, Type } from "class-transformer";


export class PatchCustomersCustomerIdUsagePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=customer_id" })
  customerId: string;
}

export class PatchCustomersCustomerIdUsageQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=timeframe_end;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  timeframeEnd: Date;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=timeframe_start;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  timeframeStart: Date;
}

export class PatchCustomersCustomerIdUsageRequestBodyEvents extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "event_name" })
  eventName: string;

  @SpeakeasyMetadata()
  @Expose({ name: "properties" })
  properties: Record<string, any>;

  @SpeakeasyMetadata()
  @Expose({ name: "timestamp" })
  timestamp: string;
}

export class PatchCustomersCustomerIdUsageRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: PatchCustomersCustomerIdUsageRequestBodyEvents })
  @Expose({ name: "events" })
  @Type(() => PatchCustomersCustomerIdUsageRequestBodyEvents)
  events: PatchCustomersCustomerIdUsageRequestBodyEvents[];
}

export class PatchCustomersCustomerIdUsageRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PatchCustomersCustomerIdUsagePathParams;

  @SpeakeasyMetadata()
  queryParams: PatchCustomersCustomerIdUsageQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PatchCustomersCustomerIdUsageRequestBody;
}

export class PatchCustomersCustomerIdUsage400ApplicationJSONValidationErrors extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "idempotency_key" })
  idempotencyKey?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "validation_errors" })
  validationErrors?: string[];
}

export class PatchCustomersCustomerIdUsage400ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status: number;

  @SpeakeasyMetadata()
  @Expose({ name: "title" })
  title: string;

  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type: string;

  @SpeakeasyMetadata({ elemType: PatchCustomersCustomerIdUsage400ApplicationJSONValidationErrors })
  @Expose({ name: "validation_errors" })
  @Type(() => PatchCustomersCustomerIdUsage400ApplicationJSONValidationErrors)
  validationErrors: PatchCustomersCustomerIdUsage400ApplicationJSONValidationErrors[];
}

export class PatchCustomersCustomerIdUsage200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "duplicate" })
  duplicate?: Record<string, any>[];

  @SpeakeasyMetadata()
  @Expose({ name: "ingested" })
  ingested?: string[];
}

export class PatchCustomersCustomerIdUsageResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  patchCustomersCustomerIdUsage200ApplicationJSONObject?: PatchCustomersCustomerIdUsage200ApplicationJSON;

  @SpeakeasyMetadata()
  patchCustomersCustomerIdUsage400ApplicationJSONObject?: PatchCustomersCustomerIdUsage400ApplicationJSON;
}